server/routes/searchDiscussions.ts

```diff
@@ -117,31 +113,13 @@ SELECT * FROM (
       "Addresses".chain as address_chain,
       "OffchainThreads".created_at,
       "OffchainThreads".chain,
-      "OffchainThreads".community
+      "OffchainThreads".community,
+      ts_rank_cd("OffchainThreads"._search, query) as rank
     FROM "OffchainThreads"
-    JOIN "Addresses" ON "OffchainThreads".address_id = "Addresses".id
-    WHERE ${communityOptions} "OffchainThreads"._search @@ plainto_tsquery('english', :searchTerm)
-    ORDER BY "OffchainThreads".created_at DESC LIMIT :limit)
-  UNION ALL
-  (SELECT
-      "OffchainThreads".title,
-      "OffchainComments".text,
-      "OffchainComments".root_id as proposalId,
-      'comment' as type,
-      "Addresses".id as address_id,
-      "Addresses".address,
-      "Addresses".chain as address_chain,
-      "OffchainComments".created_at,
-      "OffchainThreads".chain,
-      "OffchainThreads".community
-    FROM "OffchainComments"
-    JOIN "OffchainThreads" ON "OffchainThreads".id =
-        CASE WHEN root_id ~ '^discussion_[0-9\\.]+$' THEN CAST(REPLACE(root_id, 'discussion_', '') AS int) ELSE NULL END
-    JOIN "Addresses" ON "OffchainComments".address_id = "Addresses".id
-    WHERE ${communityOptions2} "OffchainComments"._search @@ plainto_tsquery('english', :searchTerm)
-    ORDER BY "OffchainComments".created_at DESC LIMIT :limit)
-) s
-ORDER BY created_at DESC LIMIT :limit;
+    JOIN "Addresses" ON "OffchainThreads".address_id = "Addresses".id, 
+    websearch_to_tsquery('english', :searchTerm) as query
+    WHERE query @@ "OffchainThreads"._search ${communityOptions} 
+    ${sort} LIMIT :limit
```

Comment:
Looks like title-matches aren't being included in Search results within a community—can we concat those in?

![image](https://user-images.githubusercontent.com/22281010/144292524-082f53fc-2276-48bf-bc49-977063f615dd.png)

Reply:
This was actually due to the migration bug earlier - the title matching wasn't applying correctly. Fixed !